### PR TITLE
chore: Add  automation test for widget isLoading property

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description
This PR adds a test for the https://github.com/appsmithorg/appsmith/pull/27406
#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/27522

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  locators,
+  entityExplorer,
+  jsEditor,
+  propPane,
+  apiPage,
+  dataManager,
+  draggableWidgets,
+} from "../../../../support/Objects/ObjectsCore";
+
+describe("Assistive Binding", function () {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts:12]: There is already a spec for AssistiveBinding inside CliendSide. Can this be moved there?

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    // Table1
+    entityExplorer.DragDropWidgetNVerify(draggableWidgets.TABLE, 100, 100);
+
+    // Button1
+    entityExplorer.DragDropWidgetNVerify(draggableWidgets.BUTTON, 200, 500);
+
+    // Text1
+    entityExplorer.DragDropWidgetNVerify(draggableWidgets.TEXT, 200, 600);
+  });
+  it("Updates isLoading property of widgets when API/Query is executing", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts:25]: nit: Add bug id to it description too for Explorer search.

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    );
+    entityExplorer.SelectEntityByName("Table1", "Widgets");
+    propPane.EnterJSContext("Table data", "{{Api1.data}}", true, false);
+
+    agHelper.AssertContains("Table1 isLoading: false");
+    agHelper.ClickButton("Submit");
+    // After triggering API execution, check that isLoading is set to true
+    agHelper.AssertContains("Table1 isLoading: true");
+    agHelper.Sleep(2000);
+    agHelper.AssertContains("Table1 isLoading: false");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts:41]: Can this validation be done in Deploy page?

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts (hunk 2 above)
Comment (ohansFavour) [reply app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts:24]: Fixed. Added to the description

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts (hunk 3 above)
Comment (ohansFavour) [reply app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts:40]: Done

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts (hunk 1 above)
Comment (ohansFavour) [reply app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts]: I have updated the description

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    // Table1
+    entityExplorer.DragDropWidgetNVerify(draggableWidgets.TABLE, 100, 100);
+
+    // Button1
+    entityExplorer.DragDropWidgetNVerify(draggableWidgets.BUTTON, 200, 500);
+
+    // Text1
+    entityExplorer.DragDropWidgetNVerify(draggableWidgets.TEXT, 200, 600);
+  });
+  it("Updates isLoading property of widgets when API/Query is executing", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts:24]: nit: Please also index the `it` blocks.

**Review by Aishwarya-U-R — COMMENTED** [review]
Approving Cypress changes.

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts (hunk 4 above)
Comment (ohansFavour) [reply app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts:24]: @Aishwarya-U-R . This has only one test. I don't think indexing is required in this case.

**Comment by ohansFavour** [comment]
> Approving Cypress changes.

This has only cypress changes. You could approve if endorsed

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts [hunk 5]
```diff
@@ -22,19 +22,8 @@ describe("Bug26935- Widget isLoading property", function () {
...
-      (req) => {
-        // Wait for 3 seconds, so we can capture the isLoading state of widgets
-        req.continue((res) => {
-          res.delay = 3000;
-          res.send();
-        });
-      },
-    );
+
+    cy.intercept("POST", "/api/v1/actions/execute", { delay: 5000 });
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug26935_spec.ts:26]: 👏

**Pull request closed without merging**